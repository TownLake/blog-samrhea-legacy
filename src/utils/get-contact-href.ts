import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
