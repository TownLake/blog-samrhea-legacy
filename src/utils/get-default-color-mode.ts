const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "light";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "light";
};

export default getDefaultColorMode;
