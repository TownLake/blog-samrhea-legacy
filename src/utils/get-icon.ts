import { ICONS } from "@/constants";

const getIcon = (name: keyof typeof ICONS) => ICONS[name] || {};

export default getIcon;
