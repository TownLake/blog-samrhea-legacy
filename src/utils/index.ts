export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
