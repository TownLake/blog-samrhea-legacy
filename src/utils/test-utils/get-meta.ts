const getMeta = (key: string) => {
  const meta = document.getElementsByTagName("meta");

  for (let i = 0; i < meta.length; i += 1) {
    if (
      key === meta[i].getAttribute("name") ||
      key === meta[i].getAttribute("property")
    ) {
      return meta[i].getAttribute("content");
    }
  }

  return "";
};

export default getMeta;
