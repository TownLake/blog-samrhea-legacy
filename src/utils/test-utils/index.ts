export {
  default as renderWithCoilProvider,
  createSnapshotsRenderer,
} from "./render-with-coil-provider";
export { default as getMeta } from "./get-meta";
