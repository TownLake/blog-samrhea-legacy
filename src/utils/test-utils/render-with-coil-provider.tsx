import React, { ReactElement } from "react";
import renderer, {
  ReactTestRenderer,
  TestRendererOptions,
} from "react-test-renderer";

import { CoilProvider } from "@alxshelepenok/diesel";
import { render, RenderOptions } from "@testing-library/react";

interface Props {
  children: React.ReactNode;
}

const WithCoilProvider: React.FC<Props> = ({ children }) => (
  <CoilProvider>{children}</CoilProvider>
);

const renderWithCoilProvider = (
  ui: React.ReactElement,
  options?: RenderOptions,
) => render(ui, { wrapper: WithCoilProvider, ...options });

export const createSnapshotsRenderer = (
  nextElement: ReactElement,
  options?: TestRendererOptions,
): ReactTestRenderer =>
  renderer.create(<CoilProvider>{nextElement}</CoilProvider>, options);

export default renderWithCoilProvider;
