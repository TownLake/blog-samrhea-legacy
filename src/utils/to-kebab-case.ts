const toKebabCase = (str: string = ""): string =>
  str
    .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
    ?.map((word) => word.toLowerCase())
    .join("-") || "";

export default toKebabCase;
