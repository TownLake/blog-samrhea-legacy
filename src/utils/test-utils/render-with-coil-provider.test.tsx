import React from "react";

import { createSnapshotsRenderer } from "@/utils/test-utils";

describe("createSnapshotsRenderer", () => {
  test("renders correctly", () => {
    const tree = createSnapshotsRenderer(<div />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
