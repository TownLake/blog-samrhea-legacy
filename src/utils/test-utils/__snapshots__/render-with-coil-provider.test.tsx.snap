// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSnapshotsRenderer renders correctly 1`] = `<div />`;
